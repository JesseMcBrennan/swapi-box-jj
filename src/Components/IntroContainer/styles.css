.target {
  position: absolute;
  z-index: -1;
  top: -1000px;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: hidden;
  font-size: 3em;
  text-align: center;
  font-family: sans-serif;
  transform-origin: 50% 100%;
  transform: perspective(300px) rotateX(19deg);
  background-color: rgb(24, 24, 24, .5);
}

.introContainer {
  padding-top: 1800px;
  animation: autoscroll 100s linear;
  color:rgba(255, 211, 0, .9)
}

.introContainer h1 {
  margin: 0;
}

@keyframes autoscroll {
  to {
    margin-top: -3000px;
  }
}
