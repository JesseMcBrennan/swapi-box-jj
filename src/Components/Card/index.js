import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

export const Card = ({name, homeworld, species, population }) => {
  return (
    <div>
      <h2>{name}</h2>
      <h2>{homeworld}</h2>
      <h2>{species}</h2>
      <h2>{population}</h2>
    </div>
  );
};

Card.propTypes = {
  name: PropTypes.string,
  homeworld: PropTypes.string,
  species: PropTypes.string,
  population: PropTypes.string
};
