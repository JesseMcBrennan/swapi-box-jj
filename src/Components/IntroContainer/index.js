import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

export const IntroContainer = ({ filmData }) => {
  const randomNumber = Math.floor(Math.random() * (7));
  const displayFilmData = filmData[`${randomNumber}`];

  return (
    <div className='target'>
      <div className="introContainer">
        <p>{displayFilmData.crawl}</p>
        <h1>{displayFilmData.title}</h1>
        <h2>{displayFilmData.releaseDate}</h2>
      </div>
    </div>
  );
}; 

IntroContainer.propTypes = {
  filmData: PropTypes.object
};