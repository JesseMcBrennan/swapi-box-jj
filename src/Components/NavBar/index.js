import React from 'react'
import './styles.css'

export const NavBar = () => {
  return(
    <div className="navBar">
      <button>People</button>
      <button>Vehicles</button>
      <button>Planets</button>
    </div>
  )
}