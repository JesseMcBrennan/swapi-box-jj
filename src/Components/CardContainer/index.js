import React from 'react';
import PropTypes from 'prop-types';
import { Card } from "../Card";
import './styles.css';

export const CardContainer = ({ peopleData, planetData, vehicleData, favoritesData }) => {
  // const displayPeopleCards = peopleData.map(person => <Card {...person} />);

  return (
    <div>
   {/*   {displayPeopleCards}*/}
    </div>
  );
};

CardContainer.propTypes = {
  peopleData: PropTypes.array
};