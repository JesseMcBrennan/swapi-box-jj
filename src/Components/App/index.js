
import React, { Component } from 'react';
import './styles.css';
import { filmDataCall } from '../../data/apiFetchCall';
import { IntroContainer } from '../IntroContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      filmData: []
    };
  }

  componentDidMount() {
    this.retrieveFilmData();
  }

  retrieveFilmData = async () => {
    const filmData = await filmDataCall();
    this.setState({
      filmData
    });
  }

  render() {
    const { filmData } = this.state;
    const displayData = filmData.length ? <IntroContainer filmData={filmData} /> : <h1>place holder</h1>;
    return (
      <div className="App">
        {displayData}
      </div>
    );
  }
}

export default App;
