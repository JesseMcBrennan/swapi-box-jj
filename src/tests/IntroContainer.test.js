import React from 'react';
import { shallow } from 'enzyme';
import { IntroContainer } from '../Components/IntroContainer';

describe('IntroContainer', () => {
  let wrapper;
  let mockFilmData;

  beforeEach(() => {
    mockFilmData = {filmInfo: 'film value'};
    wrapper = shallow(
      <IntroContainer
        filmData={mockFilmData}
      />);
  });

  it('renders the IntroContainer with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
