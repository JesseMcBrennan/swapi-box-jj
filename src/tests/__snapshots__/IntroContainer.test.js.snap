// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroContainer renders the IntroContainer with the correct props 1`] = `undefined`;
